server/src/main/java/com/cloud/user/AccountManagerImpl.java

```diff
@@ -2957,16 +2957,16 @@ public void buildACLSearchParameters(Account caller, Long id, String accountName
                 if (projectId == -1L) {
                     if (caller.getType() == Account.Type.ADMIN) {
                         domainIdRecursiveListProject.third(Project.ListProjectResourcesCriteria.ListProjectResourcesOnly);
-                        if (listAll) {
-                            domainIdRecursiveListProject.third(ListProjectResourcesCriteria.ListAllIncludingProjectResources);
-                        }
                     } else {
                         permittedAccounts.addAll(_projectMgr.listPermittedProjectAccounts(caller.getId()));
                         // permittedAccounts can be empty when the caller is not a part of any project (a domain account)
                         if (permittedAccounts.isEmpty()) {
                             permittedAccounts.add(caller.getId());
                         }
                     }
+                    if (listAll) {
+                        domainIdRecursiveListProject.third(ListProjectResourcesCriteria.ListAllIncludingProjectResources);
```

Comment:
LGTM, should we worry if there could be negative consequences? (I'm asking to understand why this case was previously suported for admin only).